### ISSUE TITLE
Xsession failed to detect zsh as login shell

### ISSUE THREAD

**Issue by cuihaoleo** [sddm/sddm#1375]
**tl;dr:** The login shell check in sddm's `Xsession` doesn't work with zsh because zsh doesn't set `SHELL` environement variable.

I set zsh as login shell, put some environment variables in `~/.profile` and sddm (plasma-x11 session) inherits those variables. Everything looks good.

However, someone points out that this shouldn't happen because sddm's `Xsession` doesn't `source ~/.profile` with zsh:
https://github.com/sddm/sddm/blob/8b371572ab6cf5f4652b7f95c700b13a89c9f109/data/scripts/Xsession#L22-L32
(Neither do my or system-wide zsh profiles `source ~/.profile`).

I put some `echo` inside `~/.profile` to figure out who reads it. It's sddm's `Xsession`:
```
cuihao      7196  0.0  0.0  16548  3760 pts/1    S+   10:30   0:00 /bin/zsh /usr/share/sddm/scripts/Xsession env
```

But, this zsh doesn't have `SHELL` environment variable set. `Xsession` go to the fallback case:
https://github.com/sddm/sddm/blob/8b371572ab6cf5f4652b7f95c700b13a89c9f109/data/scripts/Xsession#L47-L50

I can confirm that it's because `zsh` doesn't set `SHELL` if it's not set beforehand, while `bash` does (set it to the user's login shell):
```
$ env -i -- /bin/zsh -c 'echo $SHELL'

$ env -i -- /bin/bash -c 'echo $SHELL'
/bin/zsh
```

You can quickly reproduce `Xsession`'s failure like this:
```
$ env -i -- /bin/zsh -x -l /usr/share/sddm/scripts/Xsession env
......
+/usr/share/sddm/scripts/Xsession:9> case  (*/bash)
+/usr/share/sddm/scripts/Xsession:9> case  (*/zsh)
+/usr/share/sddm/scripts/Xsession:9> case  (*/csh | */tcsh)
+/usr/share/sddm/scripts/Xsession:9> case  (*/fish)
+/usr/share/sddm/scripts/Xsession:9> case  (*)
+/usr/share/sddm/scripts/Xsession:48> [ -f /etc/profile ']'
+/usr/share/sddm/scripts/Xsession:48> . /etc/profile
......
```

So `Xsession` should not rely on `SHELL` variable to check user's login shell. I think other shells' cases should be checked as well.

Environment:
* Arch Linux with latest (non-testing) packages
* Kernel: 5.11.2-arch1-1
* sddm 0.19.0

**Comment by cuihaoleo** [comment]
Well, this reminds me of another bug I submitted to Debian years ago. I am not sure if sddm people are aware of it:

[#891068 sddm should call Xsession with default shell (/bin/sh)](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=891068)

Indeed, sddm should call Xsession with bash or `/bin/sh` to avoid this kind of compatibility headache...

**Comment by cuihaoleo** [comment]
Oops, sorry. There were some errors in my debugging process.

When sddm call Xsession, `SHELL` is indeed set (by whom I don't know) so Xsession behave correctly. It was another script sourced by Xsession sourced `~/.profile`.

Anyway, above bug I sent to debian still worth a look. I am closing this one.
